lib/utils/cobs.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2024 Kelly Helmut Lord
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/net_buf.h>
+
+#include <zephyr/data/cobs.h>
+
+/**
+ * @brief COBS encode data from a net_buf into another net_buf
+ *
+ * @param src Source net_buf containing data to encode
+ * @param dst Destination net_buf to store encoded data
+ * @return int 0 on success, negative errno on failure
+ */
+int cobs_encode(struct net_buf_simple *src, struct net_buf_simple *dst)
+{
+	if (!src || !dst || !src->len) {
+		return -EINVAL;
```

Comment:
Runtime null checks aren't really the Zephyr Way.  Static analysis tools like the coverity passes we get regularly are a better paradigm for this.  You can't test for all possible programming errors.

If you feel this is a common-enough mistake to need attention here, make it an __ASSERT().

Reply:
No need for this one, but added an assert for decode.